tool/tctl/common/user_command.go

```diff
@@ -297,6 +298,14 @@ func (u *UserCommand) Add(ctx context.Context, client auth.ClientI) error {
 	user.SetTraits(traits)
 	user.SetRoles(u.allowedRoles)
 
+	// Prompt for admin action MFA if required, allowing reuse for CreateResetPasswordToken.
+	mfaResponse, err := mfa.PerformAdminActionMFACeremony(ctx, client, "CreateUser", true /*allowReuse*/)
```

Comment:
Does this mean users who do not use MFA are going to incur an additional "cost" in the form of one extra call to the cluster in order to get the MFA challenge?

I know that in the past we've been trying to avoid it. I remember Michael dealing with this around file transfer in moderated sessions. But maybe nowadays it's not that big of a deal since we want to encourage people to use MFA anyway.

Reply:
You're absolutely right. I've updated it to check the cached auth pref to see if MFA is enforced for admin actions.

> maybe we can also cache the challenge (`chal.MFARequired`)?

It's true that even with the change above we will sometimes check if MFA is required when we already know it is/is not. However, I'd like to keep this PR simple and not add a potentially finicky cache system. I'll consider making a follow up for this.